docs/modules/clients/pages/java.adoc

```diff
@@ -1866,395 +1814,330 @@ XML::
-When the iteration reaches this value and the client still could not connect to a cluster,
-it shuts down. Note that, if `try-count` was set to `1` in the above example,
-and the client could not connect to `client-config3`, it would shut down since
-it already tried once to connect to an alternative cluster.
+The following programmatic example shows how to define client using the `client-labels`
+configuration element:
 
-The following sections describe how you can configure the Java client for
-blue-green and disaster recovery scenarios.
-
-=== Configuring Using CNAME
+[source,java]
+----
+ClientConfig clientConfig = new ClientConfig();
+clientConfig.setInstanceName("ExampleClientName");
+clientConfig.addLabel("user");
+clientConfig.addLabel("bar");
 
-Using CNAME, you can change the hostname resolutions and use them dynamically.
-Let's describe the configuration with examples.
+HazelcastClient.newHazelcastClient(clientConfig);
+----
 
-Assume that you have two clusters, Cluster A and Cluster B, and two Java clients.
+For an working code sample using client labels, see the https://github.com/hazelcast/hazelcast-code-samples/tree/master/clients/client-labels[Client labels code sample].
 
-First configure the Cluster A members as shown below:
+== Query with SQL
+To query a map using SQL:
 
-[tabs] 
-==== 
-XML:: 
-+ 
--- 
-[source,xml]
+[source,java]
 ----
-<hazelcast>
-    ...
-    <network>
-        <join>
-            <tcp-ip enabled="true">
-                <member>clusterA.member1</member>
-                <member>clusterA.member2</member>
-            </tcp-ip>
-        </join>
-    </network>
-    ...
-</hazelcast>
+String query =
+        "SELECT * FROM customers csv_likes";
+try (SqlResult result = client.getSql().execute(query)) {
+    for (SqlRow row : result) {
+        System.out.println("" + row.getObject(0));
+    }
+}
 ----
---
 
-YAML::
-+
-[source,yaml]
-----
-hazelcast:
-  network:
-    join:
-      tcp-ip:
-        enabled: true
-        members: clusterA.member1,clusterA.member2
-----
-====
+For details about querying with SQL, see xref:query:sql-overview.adoc[].
 
-Then, configure the Cluster B members as shown below.
+// == Advanced configuration
+// change level
 
-[tabs] 
-==== 
-XML:: 
-+ 
--- 
-[source,xml]
-----
-<hazelcast>
-    ...
-    <network>
-        <join>
-            <tcp-ip enabled="true">
-                <member>clusterB.member1</member>
-                <member>clusterB.member2</member>
-            </tcp-ip>
-        </join>
-    </network>
-    ...
-</hazelcast>
-----
---
+== Client system properties
 
-YAML::
-+
-[source,yaml]
-----
-hazelcast:
-  network:
-    join:
-      tcp-ip:
-        enabled: true
-        members: clusterB.member1,clusterB.member2
-----
-====
+NOTE: This section is only applicable to the {java-client}.
 
-Configure the two clients as shown below.
+There are some advanced client configuration properties that help you tune the {java-client}. 
```

Comment:
The following properties are supported by {java-client-new}:

- hazelcast.client.shuffle.member.list
- hazelcast.client.heartbeat.timeout
- hazelcast.client.event.queue.capacity
- hazelcast.client.event.thread.count
- hazelcast.client.heartbeat.interval
- hazelcast.client.invocation.timeout.seconds
- hazelcast.client.invocation.retry.pause.millis
- hazelcast.client.max.concurrent.invocations
- hazelcast.client.invocation.backoff.timeout.millis
- hazelcast.client.io.input.thread.count
- hazelcast.client.io.output.thread.count
- hazelcast.client.io.balancer.interval.seconds
- hazelcast.client.io.write.through
- hazelcast.client.concurrent.window.ms
- hazelcast.client.response.thread.count
- hazelcast.client.response.thread.dynamic
- hazelcast.client.operation.backup.timeout.millis
- hazelcast.client.operation.fail.on.indeterminate.state
- hazelcast.client.statistics.period.seconds
- hazelcast.partitioning.strategy.class
- hazelcast.discovery.enabled
- hazelcast.discovery.public.ip.enabled
- hazelcast.client.connectivity.logging.delay.seconds
- hazelcast.metrics.enabled
- hazelcast.client.metrics.debug.enabled
- hazelcast.client.metrics.jmx.enabled
- hazelcast.client.metrics.collection.frequency

Reply:
Will add disclaimers for properties not supported

Reply:
Maybe we should also change the text in the beginning of the section saying that it's only applicable to the {java-client}

> == Client system properties
> 
> NOTE: This section is only applicable to the {java-client}.
> 
> There are some advanced client configuration properties that help you tune the {java-client}.